backends/ebpf/tests/ptf/meters.py

```diff
@@ -0,0 +1,157 @@
+
+
+class MeterPSATest(P4EbpfTest):
+    """
+    Test Meter used in control block. Type BYTES.
+    Send 100 B packet and verify if there is 100 tokens less left.
+    """
+
+    p4_file_path = "p4testdata/meters.p4"
+
+    def runTest(self):
+        pkt = testutils.simple_ip_packet()
+        # cir, pir -> 2 Mb/s -> 250000 byte/s, cbs, pbs -> bs (10 ms) -> 2500 B -> 09 C4
+        # period 4000 ns -> 0F A0, 1 B per period -> 01
+        self.meter_update(name="ingress_meter1", index=0,
+                          pir=250000, pbs=2500, cir=250000, cbs=2500)
+        testutils.send_packet(self, PORT0, pkt)
+        testutils.verify_packet(self, pkt, PORT1)
+        # Expecting pbs_left, cbs_left 2500 B - 100 B = 2400 B -> 09 60
+        self.verify_map_entry(name="ingress_meter1", key="hex 00",
+                              expected_value="hex "
+                                             "A0 0F 00 00 00 00 00 00 "  # pir_period
+                                             "01 00 00 00 00 00 00 00 "  # pir_unit_per_period
+                                             "A0 0F 00 00 00 00 00 00 "  # cir_period
+                                             "01 00 00 00 00 00 00 00 "  # cir_unit_per_period
+                                             "C4 09 00 00 00 00 00 00 "  # pbs
+                                             "C4 09 00 00 00 00 00 00 "  # cbs
+                                             "60 09 00 00 00 00 00 00 "  # pbs_left
+                                             "60 09 00 00 00 00 00 00 "  # cbs_left
+                                             "00 00 00 00 00 00 00 00 "  # time_p
+                                             "00 00 00 00 00 00 00 00 "  # time_c
+                                             "00 00 00 00 00 00 00 00",  # Spin lock
+                              mask=meter_value_mask)
+
+
+class MeterColorAwarePSATest(P4EbpfTest):
+    """
+    Test color-aware Meter used in control block. Type BYTES. Pre coloured with YELLOW.
+    Send 100 B packet and verify if there is 100 tokens less left in a Peak bucket only.
+    """
+
+    p4_file_path = "p4testdata/meters-color-aware.p4"
+
+    def runTest(self):
+        pkt = testutils.simple_ip_packet()
+        # cir, pir -> 2 Mb/s -> 250000 byte/s, cbs, pbs -> bs (10 ms) -> 2500 B -> 09 C4
+        # period 4000 ns -> 0F A0, 1 B per period -> 01
+        self.meter_update(name="ingress_meter1", index=0,
+                          pir=250000, pbs=2500, cir=250000, cbs=2500)
+        testutils.send_packet(self, PORT0, pkt)
+        testutils.verify_packet(self, pkt, PORT1)
+        # Expecting pbs_left, cbs_left 2500 B - 100 B = 2400 B -> 09 60
+        self.verify_map_entry(name="ingress_meter1", key="hex 00",
```

Comment:
do you want to write some utility functions to decode these values?
could be part of a future PR.
And perhaps the converse too, so you can specify the expected value in a more readable way.